package/rmfakecloud-proxy/rmfakecloudctl

```diff
@@ -0,0 +1,319 @@
+utf8 = yes
+distinguished_name = dn
+req_extensions = ext
+x509_extensions = caext
+[ dn ]
+C = AA
+ST = QQ
+L = JJ
+O  = the culture
+CN = *.appspot.com
+[ ext ]
+subjectAltName=@san
+basicConstraints=CA:FALSE
+subjectKeyIdentifier = hash
+[ caext ]
+subjectAltName=@san
+[ san ]
+DNS.1 = *.appspot.com
+DNS.2 = my.remarkable.com
+DNS.3 = internal.cloud.remarkable.com
+DNS.4 = ping.remarkable.com
+DNS.5 = *.remarkable.com
+CSR
+
+        openssl req -new -config "$csr" \
+            -key "$data_dir/rmfakecloud-proxy.key" \
+            -out "$data_dir/rmfakecloud-proxy.csr"
+        openssl x509 -req -in "$data_dir/rmfakecloud-proxy.csr" \
+            -out "$data_dir/rmfakecloud-proxy.crt" \
+            -CA "$data_dir/rmfakecloud-ca.crt" \
+            -CAkey "$data_dir/rmfakecloud-ca.key" -CAcreateserial \
+            -days 3650 -extfile "$csr" -extensions caext
+        rm -f "$data_dir/rmfakecloud-proxy.csr"
+    fi
+
+    if [[ ! -f $data_dir/rmfakecloud-proxy.bundle.crt ]]; then
+        cat "$data_dir/rmfakecloud-proxy.crt" "$data_dir/rmfakecloud-ca.crt" \
+            > "$data_dir/rmfakecloud-proxy.bundle.crt"
+    fi
+}
+
+# Untrust the local certificate authority
+uninstall-certificates() {
+    if [[ -f "$local_cert_dir/rmfakecloud-ca.crt" ]]; then
+        rm "$local_cert_dir/rmfakecloud-ca.crt"
+        update-ca-certificates --fresh
+    fi
+}
+
+# Add entries in the static host file to redirect cloud requests to the proxy
+install-hosts() {
+    uninstall-hosts
+    cat << EOF >> "$hosts_path"
+$proxy_listen hwr-production-dot-remarkable-production.appspot.com
+$proxy_listen service-manager-production-dot-remarkable-production.appspot.com
+$proxy_listen local.appspot.com
+$proxy_listen my.remarkable.com
+$proxy_listen internal.cloud.remarkable.com
+$proxy_listen ping.remarkable.com
+EOF
+}
+
+# Remove all static host entries redirecting the cloud to the proxy
+uninstall-hosts() {
+    sed -i '/ hwr-production-dot-remarkable-production.appspot.com$/d' "$hosts_path"
+    sed -i '/ service-manager-production-dot-remarkable-production.appspot.com$/d' "$hosts_path"
+    sed -i '/ local.appspot.com$/d' "$hosts_path"
+    sed -i '/ my.remarkable.com$/d' "$hosts_path"
+    sed -i '/ internal.cloud.remarkable.com$/d' "$hosts_path"
+    sed -i '/ ping.remarkable.com$/d' "$hosts_path"
+}
+
+help() {
+    echo "Usage: $(basename "$0") COMMAND
+Manage rmfakecloud-proxy. Available commands:
+
+    help                    Show this help message.
+    enable                  Enable rmfakecloud-proxy.
+    set-upstream            Define the rmfakecloud server.
+    disable                 Disable rmfakecloud-proxy."
+}
+
+if [[ $0 = "${BASH_SOURCE[0]}" ]]; then
+    if [[ $# -eq 0 ]]; then
+        help
+        exit 1
+    fi
+
+    action="$1"
+    shift
+
+    case $action in
+        enable)
+            if is-enabled; then
+                echo "rmfakecloud-proxy is already enabled."
+            else
+                disconnect-cloud
```

Comment:
Should we trap various signals here that would cause an exit and disable?

Reply:
You mean trapping script errors? Or do you have other cases in mind?

Reply:
Script errors, also ctrl-c or another interrupt signal.